File: FrontEnd/gigforce_frontend/src/services/paymentService.js
import apiClient from './apiClient';

// --- Payments Endpoints (/api/v1/payments) ---

export function getPayments() {
  return apiClient.get('/payments').then((res) => res.data); // returns List<PaymentResponseDTO>
}

export function getPaymentById(id) {
  return apiClient.get(`/payments/${id}`).then((res) => res.data);
}

export function createPayment(payload) {
  // PaymentCreateRequestDTO { invoiceId, paidAmount, paymentDate, paymentMode }
  return apiClient.post('/payments', payload).then((res) => res.data);
}

export function updatePayment(id, payload) {
  return apiClient.put(`/payments/${id}`, payload).then((res) => res.data);
}

export function processPayment(id) {
  return apiClient.put(`/payments/${id}/process`).then((res) => res.data);
}

export function failPayment(id) {
  return apiClient.put(`/payments/${id}/fail`).then((res) => res.data);
}

File: backend/src/main/java/com/gigforce/invoice/service/PaymentService.java
package com.gigforce.invoice.service;

import com.gigforce.invoice.dto.PaymentCreateRequestDTO;
import com.gigforce.invoice.dto.PaymentUpdateRequestDTO;
import com.gigforce.invoice.dto.PaymentResponseDTO;

import java.util.List;

public interface PaymentService {

    PaymentResponseDTO createPayment(PaymentCreateRequestDTO request);

    PaymentResponseDTO getPaymentById(String id);

    List<PaymentResponseDTO> getAllPayments();

    PaymentResponseDTO updatePayment(String id, PaymentUpdateRequestDTO request);

    PaymentResponseDTO processPayment(String id);

    PaymentResponseDTO failPayment(String id);
}


